File: sensors/APDS-9960/c++/gesture.cpp
#include <iostream>
#include "APDS9960.h"

using namespace std;

void handleGesture(APDS9960 apds);

int main() 
{
	APDS9960 apds = APDS9960();

	// init wiringPi
	wiringPiSetup();

	// Initialize Serial port
	cout << endl;
	cout << "------------------------------------" << endl;
	cout << " APDS-9960 - GestureTest  " << endl;
	cout << "------------------------------------" << endl;

	// Initialize APDS-9960 (configure I2C and initial values)
	if ( (false == apds.init()) || (false == apds.enableGestureSensor(false)) )
	{
		cout << "APDS-9960 init failed!" << endl;
	}

	while(1) 
	{
		handleGesture(apds);
	}
	return 0;
}

void handleGesture(APDS9960 apds) 
{
	if ( false == apds.isGestureAvailable() )
	{
		return;
	}

	switch ( apds.readGesture() ) 
	{
		case DIR_UP:
			cout << "UP" << endl;
			break;
		case DIR_DOWN:
			cout << "DOWN" << endl;
			break;
		case DIR_LEFT:
			cout << "LEFT" << endl;
			break;
		case DIR_RIGHT:
			cout << "RIGHT" << endl;
			break;
		case DIR_NEAR:
			cout << "NEAR" << endl;
			break;
		case DIR_FAR:
			cout << "FAR" << endl;
			break;
		default:
			//Print empty line if gesture in not detected
			cout << endl;
	}
}

File: sensors/APDS-9960/python/gesture.py
import os
import sys
import signal
from apds9960.const import *
from apds9960 import APDS9960
import RPi.GPIO as GPIO
import smbus
from time import sleep

def signal_handler(sig, frame):
  print("\nApplication termiated with Ctrl+C.")
  os._exit(0)

dirs = {
  APDS9960_DIR_NONE: "none",
  APDS9960_DIR_LEFT: "left",
  APDS9960_DIR_RIGHT: "right",
  APDS9960_DIR_UP: "up",
  APDS9960_DIR_DOWN: "down",
  APDS9960_DIR_NEAR: "near",
  APDS9960_DIR_FAR: "far",
}
try:
  signal.signal(signal.SIGINT, signal_handler)

  port = 1
  bus = smbus.SMBus(port)
  apds = APDS9960(bus)

  apds.setProximityIntLowThreshold(50)

  print("APDS9960 Gesture Test")
  print("============")
  apds.enableGestureSensor()

  while True:
    sleep(0.5)

    if apds.isGestureAvailable():
      motion = apds.readGesture()
      print("Gesture={}".format(dirs.get(motion, "unknown")))

except OSError:
  print('No sensor found.')
  os._exit(1)

except:
  print("Unexpected error:", sys.exc_info()[0])
  os._exit(2)

finally:
  GPIO.cleanup()
  print("Bye")
